RoaringBitmap/src/main/java/org/roaringbitmap/RoaringBitmap.java

```diff
@@ -1061,32 +1078,45 @@ public static RoaringBitmap xor(final RoaringBitmap x1, final RoaringBitmap x2)
 
   RoaringArray highLowContainer = null;
 
+  final IMemoryManager memoryManager;
+
   /**
    * Create an empty bitmap
    */
   public RoaringBitmap() {
-    highLowContainer = new RoaringArray();
+    this(DefaultMemoryManager.INSTANCE);
   }
 
+  /**
+   * Create an empty bitmap
+   * @param memoryManager the memory manager
+   */
+  public RoaringBitmap(IMemoryManager memoryManager) {
+    highLowContainer = new RoaringArray(memoryManager);
```

Comment:
How does this get freed? Whatever the mechanism, how is this ensures for existing code?

Reply:
It is freed in the clear() method, maybe we can make RoaringBitmap closable so that it frees the resources